services/common/src/redux/slices/verifiableCredentialsSlice.ts

```diff
@@ -0,0 +1,116 @@
+import { createAppSlice } from "@mds/common/redux/createAppSlice";
+import { hideLoading, showLoading } from "react-redux-loading-bar";
+import CustomAxios from "@mds/common/redux/customAxios";
+import { ENVIRONMENT, VERIFIABLE_CREDENTIALS_ISSUANCE_RECORDS } from "@mds/common/constants";
+import * as API from "@mds/common/constants/API";
+import { RootState } from "@mds/common/redux/rootState";
+
+const createRequestHeader = REQUEST_HEADER.createRequestHeader;
+
+const rejectHandler = (action) => {
+  console.log(action.error);
+  console.log(action.error.stack);
+};
+
+interface VerifiableCredentialsConnection {
+  party_guid: string;
+  permit_amendment_guid: string;
+  cred_exch_id: string;
+  cred_exch_state: string;
+  rev_reg_id: string;
+  cred_rev_id: string;
+}
+
+interface VerifiableCredentialsState {
+  verifiableCredentialConnections: VerifiableCredentialsConnection[];
+}
+
+const initialState: VerifiableCredentialsState = {
+  verifiableCredentialConnections: [],
+};
+
+const reducers = (create) => ({
+  fetchCredentialConnections: create.asyncThunk(
+    async (payload: { partyGuid: string }, thunkAPI) => {
+      const headers = createRequestHeader();
+      thunkAPI.dispatch(showLoading());
+      const { partyGuid } = payload;
+
+      const response = await CustomAxios({
+        errorToastMessage: "default",
+      }).get(`${ENVIRONMENT.apiUrl}${VERIFIABLE_CREDENTIALS_ISSUANCE_RECORDS(partyGuid)}`, headers);
+
+      thunkAPI.dispatch(hideLoading());
+
+      return response.data;
+    },
+    {
+      fulfilled: (state: RootState, action) => {
+        state.verifiableCredentialConnections = action.payload.records;
+      },
+      rejected: (state, action) => {
+        rejectHandler(action);
+      },
+    }
+  ),
+  revokeCredential: create.asyncThunk(
+    async (
+      payload: { partyGuid: string; credential_exchange_id: string; comment: string },
+      thunkAPI
+    ) => {
+      const headers = createRequestHeader();
+      thunkAPI.dispatch(showLoading());
+      const { partyGuid, credential_exchange_id, comment } = payload;
+
+      const response = await CustomAxios({
+        errorToastMessage: "default",
+      }).post(
+        `${ENVIRONMENT.apiUrl}${API.REVOKE_VERIFIABLE_CREDENTIAL(partyGuid)}`,
+        { credential_exchange_id, comment },
+        headers
+      );
+
+      thunkAPI.dispatch(hideLoading());
+
+      return response.data;
+    },
+    {
+      fulfilled: (state: RootState, action) => {
```

Comment:
oh- technically, for types here, I think that the state is actually _not_ the RootState, it's the VerifiableCredentialsState.

Reply:
I think you fixed the other instances but missed this one. 😂